lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -78,13 +78,22 @@ namespace {
       Type,
       Constructor,
       Disjunct,
-      BooleanConstant
+      BooleanConstant,
+      UnknownCase,
```

Comment:
You wound up making this a space kind anyway?  For those that want context, we discussed this in private.  The idea then was that you could traverse the pattern space to look for these - like we're already doing to find downgradable spaces - rather than using a subtraction and flattening them out.

Reply:
Ah, heh, I forgot about that conversation. I think I threw out most of it once we decided `unknown` could match patterns *containing* non-frozen enums as well, though. Do you still think I should take it out?

Reply:
The space traversal for downgradable patterns looks into nested patterns as well - unless I'm not quite understanding you here.  I haven't kept up with the Swift Evolution discussion around this all that well.

Reply:
I'll take a look tomorrow. Given that the user side of `unknown` had to be handled specially anyway, what I think you're saying might make more sense.

Reply:
…yeah, I'm not sure this is going to work. I can't figure out how to deal with this example that way:

```swift
/*non-frozen*/ public enum Test {
  case a, b, c
}

@inlinable public func consume(_ x: Test) {
  switch x {
  case .a, .b: break
  }
}
```

In this case the compiler should suggest adding both `case .c` and `unknown`. If you decompose the `Test` space, you lose the non-frozen-ness, but if you don't decompose you can't subtract out the `a` and `b` cases.

I think the other way worked when the only thing that could match future cases was irrefutable patterns, and so the suggestion for the above example could just be `default`. But now that we have to distinguish these situations I think an Unknown space does make sense.